@navikt/core/react/src/layout/grid/HGrid.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { forwardRef, HTMLAttributes } from "react";
+import cl from "clsx";
+import {
+  getResponsiveProps,
+  getResponsiveValue,
+  ResponsiveProp,
+  SpacingScale,
+} from "../utilities/css";
+
+export interface HGridProps extends HTMLAttributes<HTMLDivElement> {
+  children: React.ReactNode;
+  /**
+   * Number of columns to display. Can be a number, a string with a unit or tokens for spesific breakpoints.
+   * Sets `grid-template-columns, so 'fr, minmax' etc works`.
+   * @example
+   * columns={{ sm: 1, md: 1, lg: "1fr auto", xl: "1fr auto"}}
+   * @example
+   * columns={3}
+   * @example
+   * columns="repeat(3, minmax(0, 1fr))"
+   */
+  columns?: ResponsiveProp<number | string>;
+  /** Spacing between columns. Can be a number, a string with a unit or tokens for spesific breakpoints.
+   * @example
+   * gap="6"
+   * gap={{ sm: "2", md: "2", lg: "6", xl: "6"}}
+   */
+  gap?: ResponsiveProp<SpacingScale>;
+}
+
+export const HGrid = forwardRef<HTMLDivElement, HGridProps>(
+  ({ className, columns, gap, ...rest }, ref) => {
+    const styles = {
+      ...getResponsiveProps(`hgrid`, "gap", "spacing", gap),
+      ...getResponsiveValue(`hgrid`, "grid-columns", formatGrid(columns)),
+    } as React.CSSProperties;
```

Comment:
Ville prøvd å unngå `as`
```suggestion
    const styles: React.CSSProperties = {
      ...getResponsiveProps(`hgrid`, "gap", "spacing", gap),
      ...getResponsiveValue(`hgrid`, "grid-columns", formatGrid(columns)),
    };
```
Men usikker på om eksplisitt typing egentlig trengs her?